scipy/sparse/tests/test_coo.py

```diff
@@ -203,15 +320,44 @@ def test_2d_to_1d_resize(arg: int):
 
 
 def test_sum_duplicates():
+    # 1d case
     arr1d = coo_array(([2, 2, 2], ([1, 0, 1],)))
     assert arr1d.nnz == 3
     assert_equal(arr1d.toarray(), np.array([2, 4]))
     arr1d.sum_duplicates()
     assert arr1d.nnz == 2
     assert_equal(arr1d.toarray(), np.array([2, 4]))
 
-
-def test_eliminate_zeros():
+    # 2d case
+    arr2d = coo_array(([1, 2, 3, 4], ([0, 0, 1, 1], [1, 1, 0, 1])))
+    assert arr2d.nnz == 4
+    assert_equal(arr2d.toarray(), np.array([[0, 3], [3, 4]]))
+    arr2d.sum_duplicates()
+    assert arr2d.nnz == 3
+    assert_equal(arr2d.toarray(), np.array([[0, 3], [3, 4]]))
+
+    # 4d case
+    arr4d = coo_array(([2, 3, 7], ([1, 0, 1], [0, 2, 0], [1, 2, 1], [1, 0, 1])))
+    assert arr4d.nnz == 3
+    assert_equal(arr4d.toarray(), 
+                 np.array([[[[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0], [0, 0]],
+                          [[0, 0], [0, 0], [3, 0]]], [[[0, 0], [0, 9], [0, 0]],
+                          [[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0], [0, 0]]]]))
```

Comment:
consider writing like this to make the structure of the array clear. (Unfortunately, this still goes 1 character over our general 88 char limit, so using `  # noqa: E501` will tell the linter to allow us to break the rule here. A similar pattern could be used below.

```suggestion
    expected = np.array(  # noqa: E501
        [[[[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0], [3, 0]]],
         [[[0, 0], [0, 9], [0, 0]], [[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0], [0, 0]]]]
    )
    assert_equal(arr4d.toarray(), expected)
```